src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.spec.ts

```diff
@@ -41,174 +47,189 @@ describe('EditTechRecordButtonComponent', () => {
-
-    component.vehicleTechRecord = <VehicleTechRecordModel>{techRecord: [{statusCode: 'current', vehicleType: 'psv' }]};
-    component.viewableTechRecord = <TechRecordModel>{statusCode: 'current', vehicleType: 'psv'};
-    component.editableState = true;
-    component.isDirty = true;
-    fixture.detectChanges();
-
-    window.confirm = jest.fn(() => false);
-    jest.spyOn(component, 'cancelAmend');
-    jest.spyOn(component, 'toggleEditMode');
-
-    fixture.debugElement.query(By.css('#cancel')).nativeElement.click();
-
-    expect(component.cancelAmend).toHaveBeenCalled();
-    expect(component.toggleEditMode).toHaveBeenCalledTimes(0);
-    expect(component.editableState).toBeTruthy();
-  }))
-
-  it('should NOT promt user if cancelling an amend with a clean form.', fakeAsync(() => {
-
-    component.vehicleTechRecord = <VehicleTechRecordModel>{techRecord: [{statusCode: 'current', vehicleType: 'psv' }]}
-    component.viewableTechRecord = <TechRecordModel>{statusCode: 'current', vehicleType: 'psv'};
-
-    component.editableState = true;
-    component.isDirty = false;
-    fixture.detectChanges();
-
-    window.confirm = jest.fn(() => true);
-    jest.spyOn(component, 'cancelAmend');
-    jest.spyOn(component, 'toggleEditMode');
-
-    fixture.debugElement.query(By.css('#cancel')).nativeElement.click();
-
-    expect(component.cancelAmend).toHaveBeenCalled();
-    expect(component.toggleEditMode).toHaveBeenCalled();
-    expect(component.editableState).toBeFalsy();
-  }))
-
-  const expectedDate = new Date();
-  const expectedResult = {
-    vehicleTechRecords: [{
-      systemNumber: '1',
-      vin: '1',
-      techRecord: [{ createdAt: expectedDate } as TechRecordModel]
-    } as VehicleTechRecordModel]
-  };
-
-  it('router should be called on updateTechRecordsSuccess', fakeAsync(() => {
-    const navigateByUrlSpy = jest.spyOn(router, 'navigateByUrl').mockImplementation(() => Promise.resolve(true));
-
-    actions$.next(updateTechRecordsSuccess(expectedResult));
-
-    tick();
-
-    expect(navigateByUrlSpy).toHaveBeenCalledTimes(1);
-    expect(navigateByUrlSpy).toHaveBeenCalledWith('/tech-records/1/1/historic/' + expectedDate.getTime());
-  }));
+  })
+
+  describe('component', () => {
+    it('should create', () => {
+      expect(component).toBeTruthy();
+    });
+  })
+
+  describe('when viewing a tech record', () => {
+    it.each([
+      ['should be viewable', 'provisional', true],
+      ['should be viewable','current', true],
+      ['should not be viewable','archived', false]
+    ],)('edit button %s for %s record', (isViewable: string, statusCode: string, expected: boolean) => {
+      component.viewableTechRecord = <TechRecordModel>{ statusCode: statusCode, vehicleType: 'psv' };
+
+      fixture.detectChanges();
+  
+      const button = fixture.debugElement.query(By.css('#edit'));
+      
+      if (expected) {
+        expect(button).toBeTruthy;
+      }
+      else {
+        expect(button).toBeFalsy;
+      }
```

Comment:
Thought you had to call the functions ?  Does it work like this as well ?
```suggestion
      if (expected) {
        expect(button).toBeTruthy();
      }
      else {
        expect(button).toBeFalsy();
      }
```

Also, detail but: 

```suggestion
      expected ? expect(button).toBeTruthy() : expect(button).toBeFalsy()
```

Reply:
Agree that these need to be methods (end with brackets), but not with the second suggestion. The ternary conditional operator returns a value. In our case, we just want to call some `void` methods.

Reply:
There are loads of places where we use it that way though? such as [here](https://github.com/dvsa/cvs-app-vtm/blob/295bd1fcd4db0c9d90acf345694bbc6e30b9030b/src/app/forms/components/checkbox-group/checkbox-group.component.ts#L28) and [here](https://github.com/dvsa/cvs-app-vtm/blob/295bd1fcd4db0c9d90acf345694bbc6e30b9030b/src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts#L77). It's not because it returns something we have to do something with that return value. With this suggestion it will just execute the functions. There are examples online using ternary operators this way such as [this one](https://www.w3schools.com/react/react_es6_ternary.asp) and [this one](https://stackabuse.com/guide-to-the-ternary-operator-in-javascript/)

Reply:
Ok, nice. If W3Schools references it, that's good enough for me 🙂 Thanks for the links!

But still, this shouldn't mandate we change all our `if/else`s into ternaries. So, I'd leave this up to OP.